engine/src/main/java/org/hibernate/validator/HibernateValidatorConfiguration.java

```diff
@@ -186,4 +187,19 @@
 	 * @since 5.2
 	 */
 	HibernateValidatorConfiguration timeProvider(TimeProvider timeProvider);
+
+	/**
+	 * Registers an expression factory for use when constraint violation messages use EL.
+	 * This allows applications to avoid issues with the method {@link ExpressionFactory#newInstance()},
+	 * which uses the TCCL in some implementations.
+	 * @param expressionFactory The expression factory for use.
+	 * @return  {@code this} following the chaining method pattern
+	 * @since 5.3
+     */
+	HibernateValidatorConfiguration expressionFactory(ExpressionFactory expressionFactory);
```

Comment:
I am also not so happy about exposing `ExpressionFactory` here. It should not be required. If anything there is an issue in the implementation of the EL libs. I'd rather try to get this fixed there and in the meantime have something which "fixes" the issue under the hood. On top of things I needs fiddling with the creation of the message interpolator.

So what are the alternatives here again?